valispace/__init__.py

```diff
@@ -22,27 +22,13 @@ class API:
     ]
 
 
-    def __init__(self, url=None, username=None, password=None, keep_credentials=False, warn_https=True):
+    def __init__(self, deployment=None, username=None, password=None, keep_credentials=False, warn_https=True):
         print("\nAuthenticating Valispace...\n")
-        if url is None:
-            url = six.moves.input('Your Valispace url: ')
-
-        url = url.strip().rstrip("/")
-
-        if not (url.startswith('http://') or url.startswith('https://')):
-            url = 'https://' + url
-
-        # Check for SSL connection before sending the username and password.
-        if warn_https and url[:5] != "https":
-            sys.stdout.write("Are you sure you want to use a non-SSL connection? "
-                "This will expose your password to the network and might be a significant security risk [y/n]: ")
-            while True:
-                choice = six.moves.input().lower()
-                if choice == "y":
-                    break
-                if choice == "n":
-                    return
-                print("Please answer 'y' or 'n'")
+        
+        if deployment is None:
+            deployment = six.moves.input('Your Valispace deployment name : ')
+
+        url = "https://" + deployment + ".valispace.com"
```

Comment:
I like the idea of having just one place to set the protocol, but we cannot completely remove the option to use simple http.
Some people might use this inside a private network where https is neither required or available.
Also not always the url will be a .valispace.com one. There should still be an option to have no deployment name and ask for a full url and then the user can input what they need.

Reply:
Actually I only tested it on the demo instance, I actually didnt know if it would work on others.

I'll keep working on finding a way to centralize the url parsing without losing functionality.